src/types/redis_hash.cc

```diff
@@ -45,7 +46,13 @@ rocksdb::Status Hash::Size(const Slice &user_key, uint64_t *size) {
   HashMetadata metadata(false);
   rocksdb::Status s = GetMetadata(Database::GetOptions{}, ns_key, &metadata);
   if (!s.ok()) return s;
-  *size = metadata.size;
+  if (!metadata.IsEncodedFieldExpire()) {
+    *size = metadata.size;
+  } else {
+    std::vector<FieldValue> field_values;
+    GetAll(user_key, &field_values, HashFetchType::kOnlyKey);
```

Comment:
Would it be more effective to **count** available keys (a dedicated function) instead of loading them into memory and getting their size? If we have millions of elements, for example.

Reply:
@torwig Yes. maybe we should add a method that check if there is any unexpired field in the hash, and avoid query all the fields